scripts/git_changes_utils.py

```diff
@@ -0,0 +1,304 @@
+def get_refs() -> List[GitRef]:
+    """Returns the ref list taken from STDIN or the current branch."""
+    ref_list = []
+    if not sys.stdin.isatty():
+        # Git provides refs in STDIN.
+        ref_list = [GitRef(*ref_str.split()) for ref_str in sys.stdin]
+    if len(ref_list) == 0:
+        current_branch = common.get_current_branch_name()
+        encoded_stdout, encoded_stderr = common.start_subprocess_for_result(
+            ['git', 'show-ref', current_branch])
+        stderr = encoded_stderr.decode('utf-8')
+        if stderr:
+            raise ValueError(stderr)
+        stdout = encoded_stdout.decode('utf-8')
+        local_ref_line, remote_ref_line = stdout.splitlines()
+        local_sha, local_ref = local_ref_line.split()
+        remote_sha, remote_ref = remote_ref_line.split()
+        ref_list.append(GitRef(local_ref, local_sha, remote_ref, remote_sha))
+    if len(ref_list) > 0:
+        print('Ref list:')
+        pprint.pprint(ref_list)
+    return ref_list
+
+
+def get_changed_files(
+    ref_list: List[GitRef], remote: str
+) -> Dict[str, Tuple[List[FileDiff], List[bytes]]]:
+    """Collect modified files and ACMRT files for each branch in ref_list.
+
+    Parameter:
+        ref_list: list of references to parse (provided by git in stdin)
+        remote: str. The name of the remote being pushed to.
+
+    Returns:
+        dict. Dict mapping branch names to 2-tuples of the form (list of
+        changed files, list of ACMRT files).
+    """
+    if not ref_list:
+        return {}
+    # Avoid testing of non branch pushes (tags for instance) or deletions.
+    # TODO(#11620): Change the following to a denylist instead of an allowlist.
+    ref_heads_only = [
+        ref for ref in ref_list
+        if (ref.local_ref.startswith('refs/heads/') or ref.local_ref == 'HEAD')]
+    # Get branch name from e.g. local_ref='refs/heads/lint_hook'.
+    branches = [ref.local_ref.split('/')[-1] for ref in ref_heads_only]
+    hashes = [ref.local_sha1 for ref in ref_heads_only]
+    collected_files = {}
+    # Git allows that multiple branches get pushed simultaneously with the "all"
+    # flag. Therefore we need to loop over the ref_list provided.
+    for branch, _ in zip(branches, hashes):
+        # Get the difference to remote/develop.
+        modified_files = compare_to_remote(
+            remote, branch, remote_branch=get_parent_branch_name_for_diff())
+        acmrt_files = extract_acmrt_files_from_diff(modified_files)
+        collected_files[branch] = (modified_files, acmrt_files)
+
+    for branch, (modified_files, acmrt_files) in collected_files.items():
+        if modified_files:
+            print('\nModified files in %s:' % branch)
+            pprint.pprint(modified_files)
+            print('\nFiles with ACMRT in %s:' % branch)
+            pprint.pprint(acmrt_files)
+            print('\n')
+    return collected_files
+
+
+def get_staged_files() -> List[bytes]:
+    """Returns the list of staged files."""
```

Comment:
Note that only staged ACMRT files are returned. Maybe also change the function name to communicate this since normally "staged" can include deleted files

Reply:
Done.